### DISCUSSION TITLE
When TFT using 8-Bit mode is SPI for touchscreen possible?

### DISCUSSION THREAD

**Discussion by Harkort55** [Bodmer/TFT_eSPI#1714] [Q&A - General] [answered]
Hi All,
I am new to this TFT display (The Adafruit 2.8" and 3.2" Color TFT Touchscreen Breakout v2).  I followed the tutorials setting the display up with SPI on ESP32-WROOM-32D.  The sample text displayed on the screen but only got a disappointing 5 FPS.  Isn't SPI mode capable of faster frame rates?
Anyway, I rewired and configured to use 8-bit mode, and am getting a nice 55 FPS.

Now I am trying to get the touch part to work using XPT2046_Touchscreen library using SPI.  It's not working and when its all wired up, the TFT display goes all gray and stops displaying anything.  I am thinking that the SPI must be unavailable when the 8-bit mode is used for the display.  Is that correct?  Am I trying to use this in an unsupported way or should it be possible with this device?

Thank you for your time!
Harkort

**Comment by Bodmer** [comment, marked as answer]
If it is a ILI9341 based display you should get 25-30fps with SPI at 40MHz clock.

In parallel mode the SPI touch is not supported by the library however you can use other touch libraries like TFT_eTouch or the Adafruit ones.

**Reply by Harkort55** [reply]
Thank you very much.  I will try the other suggested libs, and then go back and try SPI again to see if I can figure out why my first attempt was so slow.  Sure appreciate you taking time to help me!
